test-data/unit/check-typeguard.test

```diff
@@ -803,3 +803,33 @@ def test() -> None:
         return
     reveal_type(x)  # N: Revealed type is "builtins.list[__main__.C]"
 [builtins fixtures/tuple.pyi]
+
+[case testTypeGuardedTypeDoesNotLeak]
+# https://github.com/python/mypy/issues/18895
+from enum import Enum
+from typing import Union
+from typing_extensions import TypeGuard, Literal
+
+class Model(str, Enum):
+    MODEL_A1 = 'model_a1'
+    MODEL_A2 = 'model_a2'
+    MODEL_B = 'model_b'
+
+MODEL_A = Literal[Model.MODEL_A1, Model.MODEL_A2]
+MODEL_B = Literal[Model.MODEL_B]
+
+def is_model_a(model: str) -> TypeGuard[MODEL_A]:
+    return True
+
+def is_model_b(model: str) -> TypeGuard[MODEL_B]:
+    return True
+
+def process_model(model: Union[MODEL_A, MODEL_B]) -> int:
+    return 42
+
+def handle(model: Model) -> int:
+    if is_model_a(model) or is_model_b(model):
+        # TODO: should start passing after #18896
```

Comment:
Sorry, I am a bit confused. Let's first work on #18896 then?

Reply:
It only removes this irrelevant error which isn't subject for this PR, but yes, agree, that'd be clearer:)